### ISSUE TITLE
model overfittlng?

### ISSUE THREAD

**Issue by 4pal** [tensorflow/nmt#366]
Noticed a strange behaviour on my tensorboard.the perplexity curve was not smooth.Was the model overfitting?i used my own dataset.how do you prepare validation and dev dataset.are they from training dataset?
![screenshot from 2018-08-02 21-08-43](https://user-images.githubusercontent.com/20737273/43602521-d3ebea5a-9698-11e8-8bd0-529d2bf7f6a2.png)

**Comment by oahziur** [comment]
train_ppl only evaluates on subset of the training data, so it will
fluctuate as it evaluate on different subset of your data at each point.

**Comment by 4pal** [comment]
thanks.How do i choose or prepare dataset?lets say i have a training set x,dev set y and test set z.ok.during draining are dev set sample and test set sample part of x that is being trained?
Separately,do you have a script to clean data,remove noise and keep urls and numbers including decimal points?
